### PR TITLE
net: wifi: Add toggle for extensions to scan specific channels

### PR DESCRIPTION
**Pull request by orrche**
The feature uses lots of RAM and breaks samples/net/wifi on ESP32. Add option to turn off this feature.

Fixes: b572e8216aa258891cb06a17b25b5770637bb337

### CODE REVIEW

File: subsys/net/l2/wifi/Kconfig [hunk 1]
```diff
@@ -95,9 +95,15 @@ config WIFI_MGMT_SCAN_MAX_BSS_CNT
 	help
 	  Maximum number of scan results to return. 0 represents unlimited number of BSSes.
 
+config WIFI_MGMT_SCAN_CHAN_SUPPORT
+	bool "Scan on specific channels support"
+	help
+	  Enabling the scanning of individual channels consumes a lot of memory.
```
Comment (krish2718) [inline subsys/net/l2/wifi/Kconfig:101]:
```suggestion
	  Enabling the scanning of individual channels. This option consumes a lot of memory.
```

**Review by krish2718 — DISMISSED** [review]

**Review by jukkar — COMMENTED** [review]
LGTM, but please squash the patches together.

File: include/zephyr/net/wifi_mgmt.h [hunk 2]
```diff
@@ -229,7 +229,9 @@ struct wifi_scan_params {
 	 *  not conforming to regulatory restrictions etc. The invoker of the API should
```
Comment (jukkar) [inline include/zephyr/net/wifi_mgmt.h:229]: The "Co-authored-by:" does not have a proper mail address, I think it should be
```
Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>
```

Comment (krish2718) [reply include/zephyr/net/wifi_mgmt.h:229]: Ah, I guess my review comment was committed and squashed from web, just removing that line is fine.

Comment (orrche) [reply include/zephyr/net/wifi_mgmt.h:231]: I see, wasn't sure what that button really did when I clicked it, removed the line now tho.

**Review by jukkar — DISMISSED** [review]
LGTM

**Review by rlubos — DISMISSED** [review]

**Comment by carlescufi** [comment]
@orrche please rebase to resolve conflicts

**Review by jukkar — APPROVED** [review]

**Comment by orrche** [comment]
Looks like the pull request #62735 is actually solving the problem I'm aiming to disable and ignore.

Closing this pull request in favor of it.

**Pull request closed without merging**